src/main/plug/sampler_kernel.cpp

```diff
@@ -670,6 +685,11 @@ namespace lsp
             afsample_t *afs     = af->vData[AFI_CURR];
             if (afs->pSource != NULL)
             {
+                afs->pSample->copy(afs->pSource);
+                nSampleRate = af->fSampleRate * 1000;
```

Comment:
Sample rate field should not be touched because it is the actual sample rate introduced by the host.